users/vieting/experiments/switchboard/transducer/feat/experiments.py

```diff
@@ -467,6 +467,97 @@ def run_mel_stage2():
     )
     return nn_system, report
 
+def run_scf_stage2():
+    gs.ALIAS_AND_OUTPUT_SUBDIR = "experiments/switchboard/transducer/feat/"
+
+    nn_system_stage1, _ = run_scf_stage1()
+    _, dev_corpora, _ = get_switchboard_data()
+    returnn_datasets = get_returnn_datasets_transducer_viterbi(keep_hashes=False)
+    returnn_args = {
+        "batch_size": 3000,
+        "datasets": returnn_datasets,
+        "extra_args": {
+            "accum_grad_multiple_step": 3,
+            "gradient_clip": 0.0,
+            "min_learning_rate": 1e-6,
+            "max_seq_length": {"classes": 600},
+        },
+        "specaug_old": {"max_feature": 15},
+        "rasr_loss_args": {"transducer_training_stage": "fullsum"},
+        "conformer_args": {"dropout": 0.25, "batch_norm_freeze": True},
+    }
+    returnn_args_keep_hash = copy.deepcopy(returnn_args)
+    returnn_args_keep_hash["extra_args"]["learning_rate_control_error_measure"] = "sum_dev_score"
+    feature_args = {
+        "class": "ScfNetwork",
+        "size_tf": 256 // 2,
+        "stride_tf": 10 // 2,
+        "preemphasis": 0.97,
+        "wave_norm": True,
+        "wave_cast": True,
+    }
+    recog_args = {
+        "lm_scales": [0.3, 0.35, 0.4, 0.45],
+        "label_scorer_args": {
+            "extra_args": {
+                "blank-label-index": 0,
+                "context-size": 1,
+                "label-scorer-type": "tf-ffnn-transducer",
+                "max-batch-size": 256,
+                "reduction-factors": 80 * 4,
+                "reduction-subtrahend": 200 - 1,  # STFT window size - 1
```

Comment:
This is log Mel specific and needs to be adapted. See my comment for `min_chunk_size` in stage 1 with SCF. Here, the subtrahend can be computed as (40 - 1) * 5 + 128 iirc. However, the RASR implementation is not based on how convolutions actually work but only allows for a single "subtrahend" not one for each layer. So we have to additionally subtract 1 to obtain 322 which should work for SCF. Please fix this and make a comment about this.

Reply:
Yes, since both features operate on 10ms level.